modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -165,6 +165,27 @@ inline void copyFromIE(const IE::Blob::Ptr &blob, MatType &mat) {
     }
 }
 
+// (taken from videoio module)
+static cv::Mutex& getInitializationMutex()
+{
+    static cv::Mutex initializationMutex;
+    return initializationMutex;
+}
+
+// (taken from DNN module)
+IE::Core& getCore(const std::string& id) {
+    static std::map<std::string, std::shared_ptr<IE::Core> > cores;
+    cv::AutoLock lock_unit(getInitializationMutex());
```

Comment:
You can have your mutex here as well, as there's no other users of this mutex in the code.

`getInitializationMutex()` is redundant.